tests/integration_tests/lightgbm_tuner_tests/test_optimize.py

```diff
@@ -0,0 +1,408 @@
+            tuner = BaseTuner(lgbm_params={'metric': metric})
+            assert not tuner.compare_validation_metrics(0.5, 0.1)
+            assert not tuner.compare_validation_metrics(0.5, 0.5)
+            assert tuner.compare_validation_metrics(0.1, 0.5)
+
+
+class TestLightGBMTuner(object):
+
+    def _helper_get_minimum_runner(self, params={}, train_set=None, kwargs_options={}):
+        # type: (Dict[str, Any], lgb.Dataset, Dict[str, Any]) -> lgb.LightGBMTuner
+
+        # Required keyword arguments.
+        dummy_dataset = lgb.Dataset(None)
+
+        kwargs = dict(
+            num_boost_round=5,
+            early_stopping_rounds=2,
+            valid_sets=dummy_dataset,
+        )
+        kwargs.update(kwargs_options)
+
+        runner = lgb.LightGBMTuner(params, train_set, **kwargs)
+        return runner
+
+    def test_no_eval_set_args(self):
+        # type: () -> None
+
+        params = {}  # type: Dict[str, Any]
+        train_set = lgb.Dataset(None)
+        with pytest.raises(ValueError) as excinfo:
+            lgb.LightGBMTuner(params,
+                              train_set,
+                              num_boost_round=5,
+                              early_stopping_rounds=2)
+
+        assert excinfo.type == ValueError
+        assert str(excinfo.value) == '`valid_sets` is required.'
+
+    def test_with_minimum_required_args(self):
+        # type: () -> None
+
+        runner = self._helper_get_minimum_runner()
+        assert 'num_boost_round' in runner.lgbm_kwargs
+        assert 'num_boost_round' not in runner.auto_options
+        assert runner.lgbm_kwargs['num_boost_round'] == 5
+
+    def test__parse_args_wrapper_args(self):
+        # type: () -> None
+
+        params = {}  # type: Dict[str, Any]
+        train_set = lgb.Dataset(None)
+        val_set = lgb.Dataset(None)
+        kwargs = dict(
+            num_boost_round=12,
+            early_stopping_rounds=10,
+            valid_sets=val_set,
+            time_budget=600,
+            best_params={},
+            sample_size=1000,
+        )
+        runner = lgb.LightGBMTuner(params, train_set, **kwargs)
+        new_args = ['time_budget', 'time_budget', 'best_params', 'sample_size']
+        for new_arg in new_args:
+            assert new_arg not in runner.lgbm_kwargs
+            assert new_arg in runner.auto_options
+
+    def test_sample_train_set(self):
+        # type: () -> None
+
+        sample_size = 3
+
+        X_trn = np.random.uniform(10, size=50).reshape((10, 5))
+        y_trn = np.random.randint(2, size=10)
+        train_dataset = lgb.Dataset(X_trn, label=y_trn)
+        runner = self._helper_get_minimum_runner(train_set=train_dataset,
+                                                 kwargs_options=dict(sample_size=sample_size))
+        runner.sample_train_set()
+
+        # Workaround for mypy.
+        if not type_checking.TYPE_CHECKING:
+            runner.train_subset.construct()  # Cannt get label before construct Dataset.
```

Comment:
```suggestion
            runner.train_subset.construct()  # Cannot get label before construct `Dataset`.
```

It seems Dataset indicates a Dataset class in lgbm, right?

Reply:
Right. `lgb.Dataset` might be better in this context. Thanks for your suggestion.

Reply:
Fixed: fc80893